### PR TITLE
chore: upgrade marked

### PR DESCRIPTION
**Pull request by oioki**
* Fixes dependabot alerts.
* TypeScript support (https://github.com/markedjs/marked/releases/tag/v6.0.0) so we no longer need `@types/marked`.

### CODE REVIEW

File: static/app/utils/marked.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import dompurify from 'dompurify';
-import marked from 'marked'; // eslint-disable-line no-restricted-imports
+import {marked, type MarkedOptions} from 'marked'; // eslint-disable-line no-restricted-imports
```
Comment (anonrig) [inline static/app/utils/marked.tsx:2]:
```suggestion
import {marked} from 'marked'; // eslint-disable-line no-restricted-imports
import type {MarkedOptions} from 'marked'; // eslint-disable-line no-restricted-imports
```

File: static/app/utils/marked.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ marked.setOptions({
     }
 
     loadPrismLanguage(lang, {
-      onLoad: () => callback?.(null, Prism.highlight(code, Prism.languages[lang], lang)),
+      onLoad: () => callback?.(null!, Prism.highlight(code, Prism.languages[lang], lang)),
```
Comment (anonrig) [inline static/app/utils/marked.tsx:65]: What is the error message without using `!`?

File: static/app/utils/marked.tsx (hunk 1 above)
Comment (anonrig) [reply static/app/utils/marked.tsx:2]: Separating `type` results in shipping less JS to client-side.

File: static/app/utils/marked.tsx (hunk 2 above)
Comment (oioki) [reply static/app/utils/marked.tsx:65]: > Argument of type 'null' is not assignable to parameter of type 'Error'.ts(2345)

Prototype of callback is
```
(parameter) callback: ((error: Error, code?: string | undefined) => void) | undefined
```

I used this answer https://stackoverflow.com/a/66994737 but without deep understanding.

Comment (anonrig) [reply static/app/utils/marked.tsx:66]: instead of null!, you can pass undefined

**Comment by anonrig** [comment]
Can you also run "yarn fix" after all required changes to fix the linting issues?

File: static/app/utils/marked.tsx (hunk 2 above)
Comment (oioki) [reply static/app/utils/marked.tsx:66]: > Argument of type 'undefined' is not assignable to parameter of type 'Error'.ts(2345)

**Review by anonrig — APPROVED** [review]

**Pull request merged**